### PR TITLE
Fix #33

### LINKED ISSUE #33: Fetch Data from Firebase Realtime Database in Descending Order[Feature Request]
Is it possible to fetch data using RealtimeDBPagination in Descending Order?

Basically I want my latest data to be on the top of the screen and older to the bottom of the screen.

A walkthrough would be appreciated.

Thanks,

### PR DESCRIPTION
**Pull request by CodeWithBishal**
Fixes #33 

Now Fetching the data from firebase realtime database by descending order is possible.

P.S. I have not tested it for all test cases, but it works pretty well for me.

### CODE REVIEW

**Review by OutdatedGuy — CHANGES_REQUESTED** [review]
Yo @CodeWithBishal, thanks for the PR. Can you revert the version update code? I'll update the version everywhere after merging some more commits.

Ping me after doing it, I'll check the working of code and merge it.

**Comment by CodeWithBishal** [comment]
Done @OutdatedGuy

File: lib/src/realtime_db_pagination.dart [hunk 1]
```diff
@@ -203,7 +209,9 @@ class _RealtimeDBPaginationState extends State<RealtimeDBPagination> {
     if (getMore) setState(() => _isFetching = true);
 
     final docsLimit = _data.length + (getMore ? widget.limit : 0);
-    var docsQuery = widget.query.limitToFirst(docsLimit);
+    var docsQuery = widget.descending
+        ? widget.query.limitToLast(docsLimit)
+        : widget.query.limitToFirst(docsLimit);
     if (_data.isNotEmpty) {
       docsQuery = docsQuery.startAt(
         Map<String, dynamic>.from(
```
Comment (OutdatedGuy) [inline lib/src/realtime_db_pagination.dart]: Do you think we also need to update the logic here?

Like `startAt` should be `endBefore` and `_data.first.value` should be `_data.last.value`?

Can you map out how the queries will work to decide if the change is needed.

File: lib/src/realtime_db_pagination.dart [hunk 2]
```diff
@@ -260,7 +268,9 @@ class _RealtimeDBPaginationState extends State<RealtimeDBPagination> {
     // To cancel previous live listener when new one is set.
     final tempSub = _liveStreamSub;
 
-    var latestDocQuery = widget.query.limitToFirst(1);
+    var latestDocQuery = widget.descending
+        ? widget.query.limitToLast(1)
+        : widget.query.limitToFirst(1);
     if (_data.isNotEmpty) {
       latestDocQuery = latestDocQuery.endBefore(
         Map<String, dynamic>.from(
```
Comment (OutdatedGuy) [inline lib/src/realtime_db_pagination.dart]: Do you think we also need to update the logic here?

Like `endBefore` should be `startAt` and `_data.first.value` should be `_data.last.value`?

Can you map out how the queries will work to decide if the change is needed.

**Review by OutdatedGuy — CHANGES_REQUESTED** [review]
Hi reviewed the PR. It looks good but I have made 2 concerns above. Can you address those?

File: lib/src/realtime_db_pagination.dart (hunk 2 above)
Comment (CodeWithBishal) [reply lib/src/realtime_db_pagination.dart]: During testing, withuout changing `_data.first.value` I tested by replacing `endBefore` with `startAt` and it did not work.

File: lib/src/realtime_db_pagination.dart (hunk 1 above)
Comment (CodeWithBishal) [reply lib/src/realtime_db_pagination.dart]: During testing, withuout changing `_data.first.value` I tested by replacing `startAt ` with `endBefore` and it did not work.

Comment (OutdatedGuy) [reply lib/src/realtime_db_pagination.dart]: Can you try all the patterns and tell me which one works correctly:

|           | **\_data.first.value** | **\_data.last.value** |
| --------- | ------------------ | ----------------- |
| **startAt**   |                    |                   |
| **endBefore** | ❌                 |                   |

File: lib/src/realtime_db_pagination.dart (hunk 2 above)
Comment (OutdatedGuy) [reply lib/src/realtime_db_pagination.dart]: Can you try all the patterns and tell me which one works correctly:

|               | **\_data.first.value** | **\_data.last.value** |
| ------------- | ---------------------- | --------------------- |
| **startAt**   | ❌                     |                       |
| **endBefore** |                        |                       |

File: lib/src/realtime_db_pagination.dart (hunk 1 above)
Comment (CodeWithBishal) [reply lib/src/realtime_db_pagination.dart]: |  | _data.first.value | _data.last.value |
| ------------- | ------------- | ------------- |
| startAt  | ✅ | ✅  |
| endBefore  | ❌  | ❌  |

File: lib/src/realtime_db_pagination.dart (hunk 2 above)
Comment (CodeWithBishal) [reply lib/src/realtime_db_pagination.dart]: |  | _data.first.value | _data.last.value |
| ------------- | ------------- | ------------- |
| startAt  | ❌ | ✅  |
| endBefore  | ✅  | ✅  |

File: lib/src/realtime_db_pagination.dart (hunk 1 above)
Comment (OutdatedGuy) [reply lib/src/realtime_db_pagination.dart]: Looks like we have to use `_data.last.value` and `endAt` for descending type.

File: lib/src/realtime_db_pagination.dart (hunk 2 above)
Comment (OutdatedGuy) [reply lib/src/realtime_db_pagination.dart]: Looks like we have to use `_data.last.value` and `startAfter` for descending type.

**Comment by OutdatedGuy** [comment]
@CodeWithBishal I have added a [commit](https://github.com/OutdatedGuy/firebase_pagination/pull/34/commits/5299603cea5b7e00b0082549bfaf415e9b4e6762) using proper queries for `descending` type. Can you confirm if those work properly?

**Comment by OutdatedGuy** [comment]
Looks like it's working fine now. I'll merge this PR now, if you find any problem please open a [new issue](https://github.com/OutdatedGuy/firebase_pagination/issues/new/choose). Thanks.

**Review by OutdatedGuy — APPROVED** [review]
LGTM!

**Pull request merged**